src/app/icd/server/tests/TestICDManager.cpp

```diff
@@ -118,1061 +122,1007 @@ class TestSubscriptionsInfoProvider : public SubscriptionsInfoProvider
     bool mHasPersistedSubscription = false;
 };
 
-class TestContext : public chip::Test::LoopbackMessagingContext
+System::Clock::Internal::MockClock * pMockClock          = nullptr;
+System::Clock::ClockBase * pRealClock                    = nullptr;
+chip::Test::LoopbackMessagingContext * pMessagingContext = nullptr;
+
+} // namespace
+
+namespace chip {
+namespace app {
+
+class TestICDManager : public ::testing::Test
 {
+
 public:
-    // Performs shared setup for all tests in the test suite
-    CHIP_ERROR SetUpTestSuite() override
+    /*
+     * Advance the test Mock clock time by the amout passed in argument
+     * and then force the SystemLayer Timer event loop. It will check for any expired timer,
+     * and invoke their callbacks if there are any.
+     *
+     * @param time_ms: Value in milliseconds.
+     */
+    static void AdvanceClockAndRunEventLoop(Milliseconds64 time)
     {
-        ReturnErrorOnFailure(LoopbackMessagingContext::SetUpTestSuite());
-        ReturnErrorOnFailure(chip::DeviceLayer::PlatformMgr().InitChipStack());
-
-        DeviceLayer::SetSystemLayerForTesting(&GetSystemLayer());
-        mRealClock = &chip::System::SystemClock();
-        System::Clock::Internal::SetSystemClockForTesting(&mMockClock);
+        pMockClock->AdvanceMonotonic(time);
+        pMessagingContext->GetIOContext().DriveIO();
+    }
 
-        return CHIP_NO_ERROR;
+    // Performs shared setup for all tests in the test suite
+    static void SetUpTestSuite()
+    {
+        if (pMockClock == nullptr)
+        {
+            pMockClock = new System::Clock::Internal::MockClock();
+            VerifyOrDie(pMockClock != nullptr);
```

Comment:
shouldn't we use ASSERT_NE instead to stay true to pw_unit format?

Reply:
Applied